client/document.go

```diff
@@ -401,16 +401,66 @@ func getDateTime(v any) (time.Time, error) {
 	return time.Parse(time.RFC3339, s)
 }
 
-func getJSON(v any) (string, error) {
-	s, err := getString(v)
-	if err != nil {
-		return "", err
+func getJSON(v any) (any, error) {
+	val, ok := v.(*fastjson.Value)
+	if !ok {
+		return v, nil
```

Comment:
todo: if `ok` is false here that suggests a programming error in which case we should not quietly ignore the value. Please either remove this check, or convert it to a nil check and test it if that is what you mean to check here.

Reply:
This func can be called with any value that's used to set document field values. Since JSON can be any of our supported scalar types we need this check.

Reply:
Same as other comment in encoding.go - please document this :)

Reply:
done